homeassistant/components/yale_smart_alarm/config_flow.py

```diff
@@ -140,35 +139,27 @@ async def async_step_user(
         )
 
 
-class YaleOptionsFlowHandler(OptionsFlow):
+class YaleOptionsFlowHandler(OptionsFlowWithConfigEntry):
```

Comment:
I'm not 100% sure about moving to `OptionsFlowWithConfigEntry`
There was an indication a while back from @MartinHjelmare that the entry is automatically available, and that `OptionsFlowWithConfigEntry` was an anti-pattern...

I think maybe we should have a helper `self._get_entry()` in the standard `OptionsFlow`, like we added `self._get_reauth_entry()` and `self._get_reconfigure_entry()` in the config flow recently.

Reply:
Maybe even add a property to "hide"  the source of the entry_id

```python
    @property
    def _entry_id(self) -> str:
        return self.handler

    @callback
    def _get_entry(self) -> ConfigEntry:
    """Return the config entry linked to the current options flow."""
    if entry := self.hass.config_entries.async_get_entry(self._entry_id):
        return entry
    raise UnknownEntry
```

Reply:
I didn't look into it but I assume given the options flow method in `ConfigFlow` it's passing the config entry to the options flow and thereby it needs to be initialized there. But I guess the base `OptionFlow` could be improved and without even a helper since an `OptionFlow` is pretty useless without a config entry.

Reply:
I read your comments again (look iing at the code). and I think that makes sense.
Should perhaps propose such change and deprecate the use of the subclass.

Reply:
I have opened the corresponding PR in https://github.com/home-assistant/core/pull/129562